interpreter/cling/lib/Utils/PlatformPosix.cpp

```diff
@@ -56,24 +57,20 @@ namespace {
     }
 
   public:
-    PointerCheck() : FD(::open("/dev/random", O_WRONLY)) {
-      if (FD == -1) ::perror("open('/dev/random')");
-    }
-    ~PointerCheck() {
-      if (FD != -1) ::close(FD);
+    PointerCheck() {
+      page_size = ::sysconf(_SC_PAGESIZE);
     }
 
     bool operator () (const void* P) {
-      if (FD == -1)
-        return false;
       // std::find is considerably slower, do manual search instead.
       if (P == lines[0] || P == lines[1] || P == lines[2] || P == lines[3]
           || P == lines[4] || P == lines[5] || P == lines[6] || P == lines[7])
         return true;
 
       // There is a POSIX way of finding whether an address
       // can be accessed for reading.
-      if (::write(FD, P, 1/*byte*/) != 1) {
+      void *base = (void *)((((const size_t)P) / page_size) * page_size);
+      if (::msync(base, page_size, 0) != 0) {
         assert(errno == EFAULT && "unexpected write error at address");
```

Comment:
We have coverage in clingtest which is not being run as part of the PRs. So merging and waiting cling testsuite to catch it has been the workflow since some time.

The assert we should have caught if we had ROOT + asserts builds as part of the PRs because I am pretty sure we have a test in roottest checking for null derefs.

Reply:
This is not a null deref, but an invalid pointer different than null. I added tests for both cases in ROOT just in case.